spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -20,33 +20,141 @@
+The Reactive Messaging specification defines a development model for declaring CDI _beans_ producing, consuming and processing messages. 
+The communication between these components uses Reactive Streams. 
 
-=== Rationale
+This specification relies on https://github.com/eclipse/microprofile-reactive-streams-operators[Eclipse MicroProfile Reactive Streams Operators] and http://cdi-spec.org/[CDI].
 
-Asynchronous communication allows temporal decoupling of services in a microservice based architecture. The implication of this temporal decoupling is that communication is allowed to occur regardless of when the parties involved in the communication are running, whether they are loaded or overloaded, and whether they are successfully processing messages or failing.
+=== Concepts
 
-In contrast, synchronous communication couples services together, binding their uptime, failure and handling of load to each other. In a chain of synchronous communication, the entire conversation will only be successful if all parties in the chain are responsive - if they are all running, processing messages successfully, and not overloaded. If just one party has a problem, all will effectively exhibit the same problem.
+This section describes the different concepts introduced by the Reactive Messaging specification
 
-In a microservice based architecture, temporal coupling results in a fragile system, with resilience and scaling properties that are worse than a monolith, hence, it is important for microservice based architectures to embrace asynchronous communication as much as possible. MicroProfile currently does not offer support for asynchronous messaging, only synchronous messaging using JAX-RS and MicroProfile REST client are supported.
+==== Overall architecture
 
-Java EE offers JMS and Message Driven Beans for handling asynchronous communication, however, there are some problems with these specifications:
+An application using Reactive Messaging is composed by CDI beans consuming, producing and processing _messages_.
 
-* Both are designed for a technology landscape where messaging was a heavy weight process used on the edge of system to hand control of a transaction from system to another, consequently there is a lot of boilerplate associated with using these technologies.
-* It is assumed their design that consistency is handled using distributed transactions. However, modern message brokers, popular in microservice deployments, such as Apache Kafka, Amazon Kinesis and Azure Event Hubs, do not support XA transactions, rather, message acknowledgement is handled using offsets with at least once delivery guarantees.
-* They do not have support for asynchronous IO, it is assumed that message processing is done on a single thread, however many modern specs are moving to asynchronous IO.
+These messages can be _internal_ messages and be received/forwared from/to different brokers.
```

Comment:
These messages can be wholly _internal_ to the application or can be sent and received via different message brokers.